programs/fund-sender/src/utils/state.rs

```diff
@@ -0,0 +1,98 @@
+    pub destination_seed: Vec<u8>,
+    // an account that can update the `destination_account` and `spend_threshold`
+    pub update_authority: Pubkey,
+    // the account of the climate token
+    pub destination_account: Pubkey,
+    // minimum threshold of yield in output_yield_account before it is allowed to send funds (in lamports)
+    pub spend_threshold: u64,
+}
+
+#[account]
+pub struct State {
+    // the state account holding all the configs from GenericStateInput and the info of total funds spent on the destination
+    pub sunrise_state: Pubkey,
+    pub update_authority: Pubkey,
+    pub destination_seed: Vec<u8>,
+    pub destination_account: Pubkey,
+    pub spend_threshold: u64,
+    pub total_spent: u64,
+    pub output_yield_account_bump: u8,
+}
+
+impl State {
+    pub fn space(len_destination_seed: u8) -> usize {
+        // find space needed for state account for current config
+        32 + 32
+            + (32 * len_destination_seed as usize)
+            + 4
+            + (len_destination_seed as usize)
+            + 32
+            + 8
+            + 8
+            + 1
+            + 8 /* Discriminator */
+    }
+}
+
+#[derive(Accounts)]
+#[instruction(sunrise_state: Pubkey, state_in: GenericStateInput)]
+pub struct RegisterState<'info> {
+    // to be used for registering state account on chain, only need to ever be done once
+    #[account(mut)]
+    pub payer: Signer<'info>,
+    #[account(
+    init,
+    space = State::space(state_in.destination_seed.len() as u8),
+    seeds = [STATE, &state_in.destination_seed, sunrise_state.key().as_ref()],
+    payer = payer,
+    bump,
+    )]
+    pub state: Account<'info, State>,
+    #[account(
+    seeds = [OUTPUT_YIELD_ACCOUNT, &state_in.destination_seed, state.key().as_ref()],
+    bump,
+    )]
+    /// CHECK: Must be correctly derived from the state
+    pub output_yield_account: UncheckedAccount<'info>,
+    pub system_program: Program<'info, System>,
+}
+
+#[derive(Accounts)]
+#[instruction(state_in: GenericStateInput)]
+pub struct UpdateState<'info> {
+    // to be used for updating parameters of state account
+    #[account(mut)]
+    pub payer: Signer<'info>,
+    #[account(
+    mut,
+    constraint = state.update_authority == payer.key() @ ErrorCode::Unauthorized,
+    )]
+    pub state: Account<'info, State>,
+    pub system_program: Program<'info, System>,
+}
+
+#[derive(Accounts)]
+#[instruction(amount: u64)]
+pub struct SendFund<'info> {
+    // to allocate correct yield proportion to various output_yield_accounts
+    #[account(mut)]
+    pub payer: Signer<'info>,
+    pub state: Account<'info, State>,
+    #[account(
+        mut,
+        seeds = [OUTPUT_YIELD_ACCOUNT, &state.destination_seed, state.key().as_ref()],
+        bump = state.output_yield_account_bump,
+    )]
+    /// CHECK: Must be correctly derived from the state
+    pub output_yield_account: UncheckedAccount<'info>,
+    pub destination_account: UncheckedAccount<'info>,
```

Comment:
We need to check that this is the correct destination account either here with something like

` #[account(
        constraint = destination_account.key() == state.destination_account
    )]`

Or as mentioned above in the instruction.